### PR TITLE
[Merged by Bors] - chore(SetTheory/Game/Nim): golf proof of Sprague-Grundy

### PR DESCRIPTION
**Pull request by vihdzp**
We also add some useful auxiliary lemmas.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Game/Nim.lean [hunk 1]
```diff
@@ -208,54 +208,44 @@ theorem nim_equiv_iff_eq {o₁ o₂ : Ordinal} : (nim o₁ ≈ nim o₂) ↔ o
 
 /-- The Grundy value of an impartial game, the ordinal which corresponds to the game of nim that the
  game is equivalent to -/
```
Comment (urkud) [inline Mathlib/SetTheory/Game/Nim.lean:210]: Please explain the recursive definition first, then mention the theorem.

**Comment by urkud** [comment]
One more comment (probably, for other PRs): `PGame.Impartial` uses an auxiliary irreducible `ImpartialAux` definition; why not use the usual pattern with a 2-fields structure (one for equiv_neg, one for the other axiom)?
LGTM. Thanks!
bors d+

**Comment by vihdzp** [comment]
I remove the `Impartial` typeclass altogether in #16054 in favor of just using a proposition with dot notation. Should I also make it into a structure there?

**Comment by urkud** [comment]
> I remove the `Impartial` typeclass altogether in #16054 in favor of just using a proposition with dot notation. Should I also make it into a structure there?

Now I see that it's a recursive definition, so `structure` may fail to work.

**Pull request closed without merging**